### ISSUE TITLE
[hold for payment 2024-12-17] [$125]  Android - Room - Room name displayed truncated on header and details when using long name

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52836]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.64-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the Expensify app.
2. Tap on FAB and select "Create Chat"
3. Tap on "Room"
4. Create a long name for the room (More than 50 characters) and save it
5. Verify that the name is displayed completely in header and on room details page

## Expected Result:
Room name should be displayed in full in header and on details page, despite using a name with more than 50 characters

## Actual Result:
Room name in header and in details is truncated when creating a long room name. (More than 50 characters)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/09c337be-5c38-4769-b8dd-07b565e3be4a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021859338322732202800</li>
        <li>Upwork Job ID: 1859338322732202800</li>
        <li>Last Price Increase: 2024-11-20</li>
<li>Automatic offers: </li>
<ul>
<li>twilight2294 | Contributor | 105029064</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by jacobkim9881** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Room title line is broken on room details page.

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/b7fdace7fd14c18e314e008554f07ad35369db6b/src/pages/ReportDetailsPage.tsx#L740

The line for room title is broken here. 

### What changes do you think we should make in order to solve the problem?
We can use `reportName` for `StringUtils.lineBreaksToSpaces(reportName)`


### What alternative solutions did you explore? (Optional)
N/A

**Comment by joekaufmanexpensify** [comment]
It is expected that we truncate in the header, but we [are supposed to](https://github.com/Expensify/App/issues/22613) show the full name in room details.

**Comment by joekaufmanexpensify** [comment]
This is a pretty minor bug, so demoting

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Room name is truncated in Room Details page
### What is the root cause of that problem?

### What changes do you think we should make in order to solve the problem?

https://github.com/Expensify/App/blob/0c99ad07fef6eb83eba46d3cab071ca395d401bf/src/pages/ReportDetailsPage.tsx#L737

here pass a new prop for   `numberOfLinesTitle`, 
like:
`  numberOfLinesTitle={5} 
`### What alternative solutions did you explore? (Optional)

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Room name is truncated in Room Details page.
### What is the root cause of that problem?
Improvement
### What changes do you think we should make in order to solve the problem?

https://github.com/Expensify/App/blob/0c99ad07fef6eb83eba46d3cab071ca395d401bf/src/pages/ReportDetailsPage.tsx#L737

here pass a new prop for   `numberOfLinesTitle`, 
like:
`  numberOfLinesTitle={5} 
`
### What alternative solutions did you explore? (Optional)

**Comment by Shahidullah-Muffakir** [comment]
> Let's go with @twilight2294's [proposal](https://github.com/Expensify/App/issues/52836#issuecomment-2489700735). We don't want any truncation so it makes sense to make the `numberOfLinesTitle` unlimited.
> 
> The first [proposal](https://github.com/Expensify/App/issues/52836#issuecomment-2489642403) has the correct RCA, but adds a limit on the `numberOfLinesTitle`. The last [proposal](https://github.com/Expensify/App/issues/52836#issuecomment-2490079508) refers to a part of the code that doesn't relate to the room title.
> 
> 🎀👀🎀 C+ reviewed

@jjcoffee I don't think passing `numberOfLinesTitle={0} ` will work, can we test it please?
and as I mentioned, I gave 5 as an example we can increase the number of lines.

**Comment by jjcoffee** [comment]
@Shahidullah-Muffakir I have tested it. You can also check out the [docs](https://reactnative.dev/docs/text#numberoflines) to see that `0` is supported.

**Comment by Shahidullah-Muffakir** [comment]
@jjcoffee , maybe I am missing something because when I pass `numberOfLinesTitle={0} `, this is the output, where the complete name the room is not shown

<img width="467" alt="Screenshot 2024-11-21 at 3 16 21 PM" src="https://github.com/user-attachments/assets/7ed1aec2-dd56-423f-aad5-0235d43d09fd">

**Comment by Shahidullah-Muffakir** [comment]
Room name has a character limit of 100. Considering this, setting `numberOfLinesTitle={5} `should be sufficient.


https://github.com/Expensify/App/blob/376f9f08755b0555cf3666c5a880542af57c7da1/src/pages/settings/Report/RoomNamePage.tsx#L73

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Android - Room - Room name displayed truncated on header and details when using long name

### What is the root cause of that problem?
We are defaulting to `numberOfLinesTitle` value of 1 as we are not passing numberOfLinesTitle prop here 
https://github.com/Expensify/App/blob/0c99ad07fef6eb83eba46d3cab071ca395d401bf/src/pages/ReportDetailsPage.tsx#L737-L738
### What changes do you think we should make in order to solve the problem?

For only room chats `isUserCreatedPolicyRoom` we should pass numberOfLinesTitle of `0`
```
                    numberOfLinesTitle={isUserCreatedPolicyRoom ? 0 : 1}

```
and set breakWord style in menu item if the numberOfLinesTitle is zero here
https://github.com/Expensify/App/blob/2d2555b173b47c0bce0b1c16b275f1f98639664c/src/components/MenuItem.tsx#L468
```
            numberOfLinesTitle === 0 && styles.breakWord,

```
We can optionally pass down the breakWord title style from the `MenuItemWithTopDescription` for cases of room or pass some boolean param `shouldBreakWord` and add the style when it is true only for our room case
### What alternative solutions did you explore? (Optional)
Optionally we can set the numberOfLinesTitle to higher value that matches the room name length limit

**Comment by FitseTLT** [comment]
@jjcoffee Can you please check my proposal you are choosing a wrong proposal that would unnecessarily break the menu truncation behaviour of other reports too like group chat. We only want to apply the changes to rooms 👍

**Comment by jjcoffee** [comment]
@FitseTLT Please explain, preferably with screenshots, in what way the proposal would break group chats.

**Comment by FitseTLT** [comment]
@jjcoffee It will remove the truncation for group chats we should apply the changes only for rooms according to https://github.com/Expensify/App/issues/52836#issuecomment-2489516532
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/6e952b44-8552-479e-bd30-f7de555ac82c">

**Comment by jjcoffee** [comment]
@joekaufmanexpensify Any thoughts on [this](https://github.com/Expensify/App/issues/52836#issuecomment-2491056626)? From the [issue](https://github.com/Expensify/App/issues/22613) you linked to:

> We'd show the [full room name in room details](https://expensify.slack.com/archives/C01GTK53T8Q/p1689002306811229?thread_ts=1686863510.616919&cid=C01GTK53T8Q), as this should be the source of truth for what the room actually is.

I would think this reasoning would apply to other areas (to have a source of truth for the full name).

**Comment by FitseTLT** [comment]
> Is the question whether we should apply the same solution to group chat names in addition to rooms names on the details page?
> 
> If so, the default group chat name is just a list of all the members, right?

@joekaufmanexpensify It can also be set to any custom name

**Comment by joekaufmanexpensify** [comment]
Yep, that makes sense. My concern is more with the default name. Group chats can be quite large. If we don't truncate the name at all and a group uses the default name and it has let's say 100 members, would we show all of them on the details page? That would be quite long... 

Whereas a custom name is going to be capped by the character limit.

**Comment by FitseTLT** [comment]
> Yep, that makes sense. My concern is more with the default name. Group chats can be quite large. If we don't truncate the name at all and a group uses the default name and it has let's say 100 members, would we show all of them on the details page? That would be quite long...
> 
> Whereas a custom name is going to be capped by the character limit.

Yep. That's why I saying we should only apply the change for rooms 👍

**Comment by Shahidullah-Muffakir** [comment]
> Yep, that makes sense. My concern is more with the default name. Group chats can be quite large. If we don't truncate the name at all and a group uses the default name and it has let's say 100 members, would we show all of them on the details page? That would be quite long...
> 
> Whereas a custom name is going to be capped by the character limit.

I think passing  `numberOfLinesTitle={5} ` will work for these cases as well.

**Comment by FitseTLT** [comment]
@shawnborton What would be your design perspective on this issue?

**Comment by shawnborton** [comment]
@FitseTLT can you summarize what decisions you need help making?

**Comment by FitseTLT** [comment]
> @FitseTLT can you summarize what decisions you need help making?

According to the OP we were aiming to display the full room name in report details page Menu item like 
<img width="669" alt="image" src="https://github.com/user-attachments/assets/c0d206e0-f29d-40fc-b670-96c922a06b61">
First do you agree with that?
Then came another point whether we should apply the same behaviour for all reports such as for group chats like
<img width="806" alt="image" src="https://github.com/user-attachments/assets/992dc69c-ae5a-4d1b-ba67-17fbcb6dc8f3">
WDYT about that too

**Comment by chiragsalian** [comment]
My opinion,
1. We should show the full room name in the details section.
2. We should not show the full group name. Group name should still be truncated. This is because group names don't have the same 100 character limit unless you edit it. Before you edit it can be very long and possibly screw up the details UI.
3. But with that said room name and group name should be of the same number of max lines so it's consistent.

Those are my thoughts but I'll let @shawnborton weigh in from a design perspective. Let me know if you've understood where we're at and if not i can summarize again.

**Comment by chiragsalian** [comment]
> if you check the video https://github.com/Expensify/App/issues/52836#issuecomment-2494174257, there is a 100 character limit for default group name as well

No i don't think that's proof. Feel free to correct me if i am mistaken. So here I have a group report with this group name,
```
chirag.exp.test1@gmail.com, chirag.exp.test1+a@gmail.com, chirag.exp.test1+b@gmail.com, chirag.exp.test1+c@gmail.com, chirag.exp.test1+d@gmail.com, chirag.exp.test1+e@gmail.com, chirag.exp.test1+f@gmail.com, chirag.exp.test1+g@gmail.com, chirag.exp.test1+h@gmail.com, chirag.exp.test1+i@gmail.com, chirag.exp.test1+j@gmail.com, chirag.exp.test1+k@gmail.com, chirag.exp.test1+l@gmail.com, chirag.exp.test1+m@gmail.com, test
```
Thats 400+ characters and i see it fully here,
![image](https://github.com/user-attachments/assets/97e7c6b5-600d-4033-b351-28ecb456ee6a)

If we show the same without a character limit in the report details screen it will be very long. Only once i edit it will it show me that i have to reduce the number of characters to 100.

**Comment by FitseTLT** [comment]
@chiragsalian We only show the first 5 emails of the participants for group chats in details page that's what is tricking @twilight2294 but that still can surpass 100 chars if the logins are very long.

**Comment by chiragsalian** [comment]
Sure, go for it. Feel free to create the PR.

**Comment by joekaufmanexpensify** [comment]
PR in review

**Comment by joekaufmanexpensify** [comment]
I see we reverted this PR and are working on a fix in https://github.com/Expensify/App/issues/53308 . I am thinking we'll handle payment once the final PR is out. Sound good @twilight2294 @jjcoffee ?

**Comment by joekaufmanexpensify** [comment]
Removed payment date, as new PR is not out to prod yet.

**Comment by joekaufmanexpensify** [comment]
This is pending https://github.com/Expensify/App/pull/53553 , which is merged but not yet out on staging.

**Comment by joekaufmanexpensify** [comment]
Second PR deployed last week. We'll handle payment here and it is due tomorrow.

**Comment by joekaufmanexpensify** [comment]
We originally worked on this bug in https://github.com/Expensify/App/pull/53041 and then reverted it because it caused a regression. We then fixed the bug again in https://github.com/Expensify/App/pull/53553. 50% regression penalty therefore applies. Payment here is as follows: 

- @twilight2294 - $62.50 for PR via upwork. 
- @jjcoffee - $62.50 for C+ via upwork.

**Comment by jjcoffee** [comment]
@joekaufmanexpensify Just to note, I get paid via ND now :pray:

**Comment by joekaufmanexpensify** [comment]
Updated. TY!

**Comment by joekaufmanexpensify** [comment]
@twilight2294 $62.50 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.

**Comment by joekaufmanexpensify** [comment]
@jjcoffee please request payment at your earliest convenience. Closing as this is otherwise all set. Thanks everyone!

**Comment by garrettmknight** [comment]
$62.50 approved for @jjcoffee
